state/convert.go

```diff
@@ -163,13 +165,42 @@ func deriveBlockHashV3(hdr flowHeader) flow.Identifier {
 	return flow.MakeID(dst)
 }
 
+func deriveBlockHashV5(hdr flowHeader) flow.Identifier {
+	dst := struct {
+		ChainID            flow.ChainID
+		ParentID           flow.Identifier
+		Height             uint64
+		PayloadHash        flow.Identifier
+		Timestamp          uint64
+		View               uint64
+		ParentView         uint64
+		ParentVoterIndices []byte
+		ParentVoterSigData []byte
+		ProposerID         flow.Identifier
+		LastViewTCID       flow.Identifier
+	}{
+		ChainID:            hdr.ChainID,
+		ParentID:           hdr.ParentID,
+		Height:             hdr.Height,
+		PayloadHash:        hdr.PayloadHash,
+		Timestamp:          uint64(hdr.Timestamp.UnixNano()),
+		View:               hdr.View,
+		ParentView:         hdr.ParentView,
+		ParentVoterIndices: hdr.ParentVoterIndices,
+		ParentVoterSigData: hdr.ParentVoterSigData,
+		ProposerID:         hdr.ProposerID,
+		LastViewTCID:       hdr.LastViewTC.ID(),
+	}
+	return flow.MakeID(dst)
+}
+
 func deriveEventsHash(spork *config.Spork, events []flowEvent) flow.Identifier {
 	switch spork.Version {
 	case 1:
 		return deriveEventsHashV1(events)
 	case 2, 3:
 		return deriveEventsHashV2(events)
-	case 4:
+	case 4, 5:
```

Comment:
v4 and v5 deriveEventsHash is the same?

Reply:
Yep, the only thing we wanted to update was the block header hashing in the deriveBlockHash function